include/ozo/impl/connection.h

```diff
@@ -85,7 +85,7 @@ void connection<OidMap, Statistics>::cancel() noexcept {
 
 template <typename OidMap, typename Statistics>
 bool connection<OidMap, Statistics>::is_bad() const noexcept {
-    return detail::connection_status_bad(native_handle());
+    return !native_handle() || PQstatus(native_handle()) != CONNECTION_OK;
```

Comment:
IMO it is better to make a complementary function `detail::connection_status_ok()` in the same way that `detail::connection_status_bad()` implemented and use it negation in both places to preserve the logic and semantic if you do not mind.

Reply:
Ok, but if we are going to implement `detail::connection_status_ok()` , it should be like this: `native_handle() && PQstatus(native_handle()) == CONNECTION_OK` , and its negation would not be what we guess

Reply:
> its negation would not be what we guess

Well, as for me, everything is ok:
`native_handle() && PQstatus(native_handle()) == CONNECTION_OK`
`!(native_handle() && PQstatus(native_handle()) == CONNECTION_OK)`
`!native_handle() || !(PQstatus(native_handle()) == CONNECTION_OK)`
`!native_handle() || PQstatus(native_handle()) != CONNECTION_OK`

Am I wrong here?

Reply:
No, now i see, thx